github/repos.go

```diff
@@ -1037,3 +1037,38 @@ func (s *RepositoriesService) ReplaceAllTopics(ctx context.Context, owner, repo
 
 	return t.Names, resp, nil
 }
+
+// TransferRequest represents a request to transfer a repository.
+type TransferRequest struct {
+	NewOwner string `json:"new_owner,omitempty"`
+	TeamID   []int  `json:"team_id,omitempty"`
+}
+
+// Transfer transfers a repository from one account or organization to another.
+//
+// This method might return an *AcceptedError and a status code of
+// 202. This is because this is the status that GitHub returns to signify that
+// it has now scheduled the transfer of the repository in a background task.
+// A follow up request, after a delay of a second or so, should result
+// in a successful request.
+//
+// GitHub API docs: https://developer.github.com/v3/repos/#transfer-a-repository
+func (s *RepositoriesService) Transfer(ctx context.Context, owner, repo string, transfer *TransferRequest) (*Repository, *Response, error) {
```

Comment:
Why is `transfer` a `*TransferRequest` pointer rather than value?

That means user can pass `nil`, but that wouldn't be meaningful, would it?

Reply:
I understand your point here. And while I tend to agree, there's precedent for this approach even in repos.create being passed a pointer to a Repo. I didn't want to introduce code that broke with that model.

Thoughts?

Reply:
> there's precedent for this approach even in repos.create being passed a pointer to a Repo. I didn't want to introduce code that broke with that model.

I see. However, that API decision was made [5 years ago](https://github.com/google/go-github/commit/d3d9655fad081711f66546dbd439814a8ef1d5c7#diff-49ba3de83fa99f3d8bae2ece519a449fR153), and is not very representative of the best practices regarding API decisions this repo has evolved toward since that time.

In fact, we've ran into possible panics in the library caused due to such parameters being passed as pointers in situations where nil values made no sense. We dealt with that in #531 and #539, by adding `if param == nil { return errors.New(...) }` guards.

We went that route because it was less disruptive than breaking an API for a very minor improvement. Given two equivalent choices, we prefer consistency.

However, in this case, it's a new API, so we can go with the best API design we can come up with. Using a pointer isn't meaningful and doesn't add value (does it?), but it create a liability for dealing with users potentially passing `nil` values.

I understand your point about wanting to be consistent with existing code, but in this case, I don't think the consistency argument applies, because using a pointer has tangible downsides compared to using a value. They're not equivalent choices. When adding new APIs to this library, we try to follow the best known practices, and we sometimes even break APIs of preview endpoints when improvements are discovered.

If you want to see more recent precedent, see #773 which came up during review of #771. The original version started by using pointers, but we came to agreement that value parameters were better.

(One could argue that `TransferRequest` contains just 2 fields that could be inlined into the method signature. However, I think using a dedicated struct for it as you've done makes sense, because it maps more closely to GitHub API as [documented](https://developer.github.com/v3/repos/#input-3), and will be easy to update should GitHub add extra fields to that input object.)

What do you think?

Reply:
That sounds reasonable to me. Thanks for the thorough background! I'll go ahead and work on the updates today.

I had thought about your parenthetical initially, but decided against it for exactly the reasons you enumerated.